sematic/ui/src/components/Docstring.tsx

```diff
@@ -19,7 +19,7 @@ export default function Docstring(props: {
       }}
     >
       {(docstring !== undefined && docstring !== null && (
-        <ReactMarkdown>{docstring}</ReactMarkdown>
+        <ReactMarkdown className="line-break">{docstring}</ReactMarkdown>
```

Comment:
Is it possible to actually render it as Markdown does?

Reply:
Of course, we can replace the string into correct Markdown format. The [cited link](https://github.com/remarkjs/react-markdown/issues/273) mentioned several ways to do that. It might involve more slightly more code though. Do you prefer that way?

Reply:
Yes, please do try.

Reply:
PTAL

Reply:
It seems some empty lines are missing, and one is added, somehow. You also mentioned that trailing whitespaces are removed upstream.
![image](https://user-images.githubusercontent.com/1894533/201976729-0530c181-63b1-4b7d-b6a6-cddc8b87a1a4.png)

Can you please investigate the causes? It would be very user-friendly to display the docstrings like in an IDE.

Reply:
Yes, you are right, some lines are indeed missing. 

> One is added
I am testing the result, so I intentionally make multiple blank lines like this:

<img width="892" alt="image" src="https://user-images.githubusercontent.com/1046489/201990410-81232320-1b2e-4596-9f69-f57efb3d97c7.png">

So it is not added automatically. Sorry for the confusion.

> You also mentioned that trailing whitespaces are removed upstream.

Actually I mentioned about leading whitespaces, not the trailing whitespaces. 

Now I think if a docstring is written as a plain text, perhaps we should not let it render as Markdown or there will always be gaps. Using `<pre>` to display it might be a better fit. Can we agree on this implementation instead?

Whist there are still other issues, like the leading whitespaces removal. I will investigate further.

Reply:
We do indeed trim leading whitespaces on the backend. See https://github.com/sematic-ai/sematic/blob/9dd5126861f8071c9968c1eef6ab080c1c5228cd/sematic/db/models/run.py#L144-L149

The idea was to enable markdown but not take the indentation into account, otherwise markdown would not work.

I still think it's quite valuable to support Markdown since it can include links, structured texts with headers and lists, etc.

The issue here is that our Docstring are not written in markdown, they are written in [Numpy style](https://betterprogramming.pub/3-different-docstring-formats-for-python-d27be81e0d68). So it is the expected behavior that they do not render correctly.

We should have a principled approach to this. A few options:

- we support markdown exclusively. If a docstring is not markdown, then 🤷 we do a best effort. Then we also update all our Sematic function docstrings to be markdown.
- we support raw text only with `pre`, then we lose links, headers, lists, etc
- we find a way to tell if a text is markdown or not (either by parsing, or by having the user set a flag somewhere in the docstring) and use either markdown or `pre` accordingly.

Reply:
Many thanks @neutralino1  !

I incline to do #1 `support markdown exclusively` for now. But I also want to expand a little bit on #3. `set a flag somewhere in the docstring` so is that to use a shebang? something like:

```
"""MD
     Attention: here is a docstring for the below function.
"""
```
(MD itself supports this syntax for code sections)

Coveats/concenrs: 1. Would it be anti-pythonic and make python users uncomfortable? since we are inventing something heretic. 
2. If we decide to do so, let's make it a feature explicit to the users. Let's add it to the doc.

One last thought: An alternative to shebang is to use another py annotation, telling the docstring of the associated function is a Markdown.

Reply:
I am in favor of sticking to # 1 as it is closest to what we currently do. I want us to be careful of not adding too many features.
In that case, it seems the only thing to do is to update our docstrings for pipeline functions (not other functions in the codebase) to be markdown exclusively.

Reply:
Thanks. I updated the docstring of the examples. There are two examples that require Markdown improvement. 

PTAL.